internal/controller/replication.storage/status.go

```diff
@@ -46,26 +49,30 @@ func setPromotedCondition(conditions *[]metav1.Condition, observedGeneration int
 }
 
 // sets conditions when volume promotion was failed.
-func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGeneration int64) {
+func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGeneration int64, errMessage string, errFromCephCSI string) {
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            errMessage,
 		Type:               v1alpha1.ConditionCompleted,
 		Reason:             v1alpha1.FailedToPromote,
 		ObservedGeneration: observedGeneration,
 		Status:             metav1.ConditionFalse,
 	})
 	setStatusCondition(conditions, &metav1.Condition{
+		Message:            v1alpha1.MessagePromotionErrorDetected,
 		Type:               v1alpha1.ConditionDegraded,
-		Reason:             v1alpha1.Error,
+		Reason:             errFromCephCSI,
```

Comment:
types.Condition is documenting the Reason and Message:

```go
  	// reason contains a programmatic identifier indicating the reason for the condition's last transition.
	// Producers of specific condition types may define expected values and meanings for this field,
	// and whether the values are considered a guaranteed API.
	// The value should be a CamelCase string.
	// This field may not be empty.
	// +required
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MaxLength=1024
	// +kubebuilder:validation:MinLength=1
	// +kubebuilder:validation:Pattern=`^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$`
	Reason string `json:"reason" protobuf:"bytes,5,opt,name=reason"`
	// message is a human readable message indicating details about the transition.
	// This may be an empty string.
	// +required
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MaxLength=32768
	Message string `json:"message" protobuf:"bytes,6,opt,name=message"`
```

Using Reason for the details of the message contradicts these docs. This means that our API will not behave like most other APIs, and will be harder to use.

This is also harder to use - ramen will have to read both .Mesage and .Reason, and construct a new mesage from both, handling all cases (message empty, reason empty, both empty, ...). It is much easier to use when we can copy the message from the condition as is.

I think we should do instead:

```go
    Message: fmt.Sprintf("Failed to promote: %s", errFromCephCSI)
```

The "Failed to promote: %s" part can be a constant, helper developers use consistent messages and making changes easy. 

See also https://github.com/evankanderson/k8s-community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties